packages/contracts/test/forwarder/forwarder.ts

```diff
@@ -0,0 +1,306 @@
+            from: owner,
+        });
+
+        defaultMakerAssetAddress = erc20TokenA.address;
+        defaultTakerAssetAddress = etherTokenInstance.address;
+        const defaultOrderParams = {
+            exchangeAddress: exchangeInstance.address,
+            makerAddress,
+            feeRecipientAddress,
+            makerAssetData: assetProxyUtils.encodeERC20ProxyData(defaultMakerAssetAddress),
+            takerAssetData: assetProxyUtils.encodeERC20ProxyData(defaultTakerAssetAddress),
+            makerAssetAmount: ZeroEx.toBaseUnitAmount(new BigNumber(200), DECIMALS_DEFAULT),
+            takerAssetAmount: ZeroEx.toBaseUnitAmount(new BigNumber(10), DECIMALS_DEFAULT),
+            makerFee: ZeroEx.toBaseUnitAmount(new BigNumber(1), DECIMALS_DEFAULT),
+            takerFee: ZeroEx.toBaseUnitAmount(new BigNumber(0), DECIMALS_DEFAULT),
+        };
+        const privateKey = constants.TESTRPC_PRIVATE_KEYS[accounts.indexOf(makerAddress)];
+        orderFactory = new OrderFactory(privateKey, defaultOrderParams);
+
+        const forwarderArgs = [
+            exchangeInstance.address,
+            etherTokenInstance.address,
+            zrxToken.address,
+            AssetProxyId.ERC20,
+        ];
+        const forwarderInstance = await deployer.deployAndSaveAsync(ContractName.Forwarder, forwarderArgs);
+        forwarderContract = new ForwarderContract(forwarderInstance.abi, forwarderInstance.address, provider);
+        await forwarderContract.setERC20ProxyApproval.sendTransactionAsync(AssetProxyId.ERC20, { from: owner });
+        forwarderWrapper = new ForwarderWrapper(forwarderContract, web3Wrapper);
+        erc20Wrapper.addTokenOwnerAddress(forwarderInstance.address);
+
+        web3Wrapper.abiDecoder.addABI(forwarderContract.abi);
+        web3Wrapper.abiDecoder.addABI(exchangeInstance.abi);
+    });
+    beforeEach(async () => {
+        await blockchainLifecycle.startAsync();
+        erc20Balances = await erc20Wrapper.getBalancesAsync();
+        signedOrder = orderFactory.newSignedOrder();
+        feeOrder = orderFactory.newSignedOrder({
+            makerAssetData: assetProxyUtils.encodeERC20ProxyData(zrxToken.address),
+            takerFee: ZeroEx.toBaseUnitAmount(new BigNumber(1), DECIMALS_DEFAULT),
+        });
+        orderWithFees = orderFactory.newSignedOrder({
+            takerFee: ZeroEx.toBaseUnitAmount(new BigNumber(1), DECIMALS_DEFAULT),
+        });
+    });
+    afterEach(async () => {
+        await blockchainLifecycle.revertAsync();
+    });
+    describe('buyTokens', () => {
+        it('should fill the order', async () => {
+            const fillAmount = signedOrder.takerAssetAmount.div(2);
+            const makerBalanceBefore = erc20Balances[makerAddress][defaultMakerAssetAddress];
+            const takerBalanceBefore = erc20Balances[takerAddress][defaultMakerAssetAddress];
+            const tx = await forwarderWrapper.buyTokensAsync([signedOrder], [], fillAmount, takerAddress);
```

Comment:
Assign `[signedOrder]` and `[]` to well-named variable names to help code readability.